### PR TITLE
Bd/add map points

### PR DESCRIPTION
**Pull request by bdavis3**
Allow new water sources to be added

### CODE REVIEW

**Comment by abailey96** [comment]
Can you rename this PR to be more descriptive. Something like "Allow water sources to be added"?

File: src/main/java/controller/MainController.java [hunk 1]
```diff
@@ -69,6 +76,19 @@ private void onMapViewInitialized() {
             map.addMarker(marker);
             map.addUIEventHandler(marker, UIEventType.click, obj -> onWaterSourceMarkerClick(source));
         }
+
+        map.addUIEventHandler(UIEventType.click, this::addPoint);
+
+    }
+
+
+    private void addPoint(JSObject obj) {
```
Comment (abailey96) [inline src/main/java/controller/MainController.java]: Remove the empty space before this and maybe add a Javadoc.

File: src/main/java/controller/NewWaterSourceController.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import repository.WaterSourceStore;
+import util.NavigationUtilities;
+import model.WaterSource;
+
+import javax.inject.Inject;
+import java.util.Set;
+
+
+
+/**
```
Comment (abailey96) [inline src/main/java/controller/NewWaterSourceController.java]: Remove two of the empty lines before this javadoc

File: src/main/java/controller/NewWaterSourceController.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+     * Called upon initial scene load.
+     */
+    @FXML
+    public void initialize() {
+        Injector.component().inject(this);
+        type.setItems(FXCollections.observableArrayList(WaterType.values()));
+        condition.setItems(FXCollections.observableArrayList(Condition.values()));
+    }
+
+    public void setLatLong(double lat, double lng) {
```
Comment (abailey96) [inline src/main/java/controller/NewWaterSourceController.java]: Javadoc this

File: src/main/java/controller/NewWaterSourceController.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        this.lat = lat;
+        this.lng = lng;
+    }
+
+
+    public void exit() {
```
Comment (abailey96) [inline src/main/java/controller/NewWaterSourceController.java]: Should this be public? This doesn't look like it should be public

File: src/main/java/controller/NewWaterSourceController.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        NavigationUtilities.changeScreen("screen_main");
+    }
+
+    public void add() {
```
Comment (abailey96) [inline src/main/java/controller/NewWaterSourceController.java]: Consider renaming this to "commitSource" or "createSourceAndClose" or something to be more descriptive.

Comment (abailey96) [inline src/main/java/controller/NewWaterSourceController.java]: This should also be private

File: src/main/java/controller/NewWaterSourceController.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        WaterReport newReport = new WaterReport(newSource, mgr.getUser().getName(), condition.getValue(), type.getValue());
+        reportStore.add(newReport);
+
+        NavigationUtilities.changeScreen("screen_main");
+    }
+
+
+
+
+
```
Comment (abailey96) [inline src/main/java/controller/NewWaterSourceController.java]: Remove all but one of these blank lines

File: src/main/java/controller/NewWaterSourceController.java (hunk 5 above)
Comment (bdavis3) [inline src/main/java/controller/NewWaterSourceController.java]: it didn't work when i tried making these private

**Comment by abailey96** [comment]
:+1: looks good. Merging!

**Pull request merged**